libs/blocks/quiz-entry/quiz-entry.js

```diff
@@ -73,127 +116,150 @@ const App = ({
+        userFlow: nextFlow,
+        userSelection: nextSelections,
+      };
+      localStorage.setItem('stored-quiz-state', JSON.stringify(currentQuizState));
+      setQuizState(currentQuizState);
 
-  const onQuizButton = async () => {
-    const { mlDetails, mlValues } = mlData;
-    const mlFieldText = document.querySelector('#ml-field-input').value;
-    const resultContainer = document.querySelector('.results-container');
-    if (mlFieldText.length > 0) {
-      const fiResults = await getFiResults(mlDetails.endpoint, mlDetails['api-key'], mlFieldText, 10, mlValues);
-      const { data } = fiResults;
-      resultContainer.replaceChildren();
-      data.forEach((value) => {
-        const fiCode = createTag('p', '', `ficode: ${value.ficode} | prob: ${value.prob}`);
-        resultContainer.appendChild(fiCode);
-      });
-    } else {
-      resultContainer.replaceChildren();
+      // eslint-disable-next-line no-console
+      if (debug) console.log(currentQuizState);
+      if (!debug) window.location = quizPath;
     }
-    // console.log('quizPath', quizPath);
   };
 
   const onOptionClick = (option) => () => {
-    const newState = { ...selectedCards };
+    const selected = { ...selectedCards };
+    const selectedTotal = Object.keys(selected).length;
 
-    if (Object.keys(newState).length >= maxSelections && !newState[option.options]) {
+    if (selectedTotal >= maxSelections && !selected[option.options]) {
       return;
     }
 
-    if (!newState[option.options]) {
-      newState[option.options] = true;
+    if (!selected[option.options]) {
+      selected[option.options] = true;
     } else {
-      delete newState[option.options];
+      delete selected[option.options];
     }
 
-    setSelectedCards(newState);
-    setCountOfSelectedCards(Object.keys(newState).length);
+    setSelectedCards(selected);
   };
 
+  const onMLInput = (event) => {
+    const inputValue = event.target.value;
+    setMLInputUsed(true);
+    if (inputValue.length === 0) {
+      setMLInputUsed(false);
+    }
+  };
+
+  const onMLEnter = (event) => {
+    if (event.keyCode === enterKeyCode) continueQuiz();
+  };
+
+  if (selectedQuestion) {
+    maxSelections = +selectedQuestion['max-selections'];
+  }
+
   if (!dataLoaded || !selectedQuestion) return null;
+
   return html`<div class="quiz-entry-container">
     <div class="quiz-entry-title">${quizLists.strings[selectedQuestion.questions].heading}</div>
     <div class="quiz-entry-subtitle">${quizLists.strings[selectedQuestion.questions]['sub-head']}</div>
-    ${hasMLData && html`<${mlField} cardsUsed="${cardsUsed}" onMLInput="${onMLInput}" placeholderText="${getOptionsValue('fi_code', 'title')}"/><div class="results-container"></div>`}
+    ${hasMLData && html`<${mlField} 
+      cardsUsed="${cardsUsed}" 
+      onMLInput="${onMLInput}"
+      onMLEnter="${onMLEnter}"
+      placeholderText="${getOptionsValue('fi_code', 'title')}"/><div class="results-container"></div>`}
```

Comment:
We don't need results-container

Reply:
Resolving this since you agreed to remove the extraneous markup in your analytics PR.